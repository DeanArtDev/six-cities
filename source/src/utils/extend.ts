const extend = (a: object, b: object): object => {
  const emptyObj: object = {};

  return Object.assign(emptyObj, a, b);
};

export default extend;
