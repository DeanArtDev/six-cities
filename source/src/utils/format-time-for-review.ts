import {MONTHS} from '@root/consts/consts';

const formatTimeForReview = (date: string): string => {
  const nextDate = new Date(date);
  const month = MONTHS[nextDate.getMonth()];
  const day = nextDate.getDate() < 10 ? `0${nextDate.getDate()}` : nextDate.getDate();

  return `${month} ${day} ${nextDate.getFullYear()}`;
};

export default formatTimeForReview;
