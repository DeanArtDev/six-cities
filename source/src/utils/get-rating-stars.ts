const ONE_STAR_PRESENTS = 20;

const getRatingStars = (value: number): number | 0 => {
  if (value === 0 || value === undefined || isNaN(value)) {
    return 0;
  }

  return Math.ceil(value) * ONE_STAR_PRESENTS;
};

export default getRatingStars;
