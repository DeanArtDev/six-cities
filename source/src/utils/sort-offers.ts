import {OffersSortTypes} from '@root/types';

const sortOffers = (typeSort, offers) => {
  switch (typeSort) {
    case OffersSortTypes.LOW_TO_HIGH:
      return offers
        .slice()
        .sort((max, min) => max.price - min.price);

    case OffersSortTypes.HIGH_TO_LOW:
      return offers
        .slice()
        .sort((max, min) => min.price - max.price);

    case OffersSortTypes.RATED_FIRST:
      return offers
        .slice()
        .sort((max, min) => min.rating - max.rating);

    default:
      return offers;
  }
};

export default sortOffers;
