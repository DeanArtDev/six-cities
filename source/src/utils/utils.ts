export {default as getRatingStars} from './get-rating-stars';
export {default as extend} from './extend';
export {default as sortOffers} from './sort-offers';
export {default as formatTimeForReview} from './format-time-for-review';
export {default as noop} from './noop';
