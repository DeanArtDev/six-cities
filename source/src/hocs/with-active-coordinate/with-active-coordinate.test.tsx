import withActiveCoordinate from './with-active-coordinate';
import * as React from 'react';
import * as renderer from 'react-test-renderer';
import {noop} from '@root/utils/utils';

describe(`<withActiveCoordinate/>: `, () => {

  const MockComponent = () => (
    <div/>
  );

  const MockWrapper = withActiveCoordinate(MockComponent);

  it(`should render correctly`, () => {
    const wrapper = renderer
      .create(
          <MockWrapper
            activeCoordinate={noop}
            onChangerOfferCoordinate={noop()}
          />
      ).toJSON();

    expect(wrapper).toMatchSnapshot();
  });
});
