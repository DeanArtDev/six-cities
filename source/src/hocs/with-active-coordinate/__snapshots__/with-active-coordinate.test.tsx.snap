// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<withActiveCoordinate/>:  should render correctly 1`] = `<div />`;
